cpp/src/arrow/engine/substrait/serde.h

```diff
@@ -36,193 +36,193 @@ namespace arrow {
-/// \return the corresponding Arrow compute expression
+/// surrounding Plan message.
+/// \return the corresponding Arrow compute expression.
 ARROW_ENGINE_EXPORT
 Result<compute::Expression> DeserializeExpression(const Buffer& buf,
                                                   const ExtensionSet& ext_set);
 
-/// \brief Serializes an Arrow compute expression to a Substrait Expression message
+/// \brief Serializes an Arrow compute expression to a Substrait Expression message.
 ///
-/// \param[in] expr the Arrow compute expression to serialize
+/// \param[in] expr the Arrow compute expression to serialize.
 /// \param[in,out] ext_set the extension mapping to use; may be updated to add
-/// mappings for the types used in the expression
+/// mappings for the types used in the expression.
 /// \return a buffer containing the protobuf serialization of the corresponding Substrait
-/// Expression message
+/// Expression message.
 ARROW_ENGINE_EXPORT
 Result<std::shared_ptr<Buffer>> SerializeExpression(const compute::Expression& expr,
                                                     ExtensionSet* ext_set);
 
-/// \brief Deserializes a Substrait Rel (relation) message to an ExecNode declaration
+/// \brief Deserializes a Substrait Rel (relation) message to an ExecNode declaration.
 ///
 /// \param[in] buf a buffer containing the protobuf serialization of a Substrait
-/// Rel message
+/// Rel message.
 /// \param[in] ext_set the extension mapping to use, normally provided by the
-/// surrounding Plan message
-/// \return the corresponding ExecNode declaration
+/// surrounding Plan message.
+/// \return the corresponding ExecNode declaration.
 ARROW_ENGINE_EXPORT Result<compute::Declaration> DeserializeRelation(
     const Buffer& buf, const ExtensionSet& ext_set);
 
 namespace internal {
 
 /// \brief Checks whether two protobuf serializations of a particular Substrait message
-/// type are equivalent
+/// type are equivalent.
 ///
 /// Note that a binary comparison of the two buffers is insufficient. One reason for this
 /// is that the fields of a message can be specified in any order in the serialization.
 ///
-/// \param[in] message_name the name of the Substrait message type to check
-/// \param[in] l_buf buffer containing the first protobuf serialization to compare
-/// \param[in] r_buf buffer containing the second protobuf serialization to compare
-/// \return success if equivalent, failure if not
+/// \param[in] message_name the name of the Substrait message type to check.
+/// \param[in] l_buf buffer containing the first protobuf serialization to compare.
+/// \param[in] r_buf buffer containing the second protobuf serialization to compare.
+/// \return success if equivalent, failure if not.
 ARROW_ENGINE_EXPORT
 Status CheckMessagesEquivalent(util::string_view message_name, const Buffer& l_buf,
                                const Buffer& r_buf);
 
 /// \brief Utility function to convert a JSON serialization of a Substrait message to
```

Comment:
Or if we want to keep it concise
```suggestion
/// \brief Convert a JSON serialization of a Substrait message to
```

Reply:
6d61c85